### DISCUSSION TITLE
How to use nested include?

### DISCUSSION THREAD

**Discussion by BJladika** [prisma/prisma#7551] [Q&A]
Okay, I'm now testing nested include, like 
```graphql
query{
  students2{
    Name
    Status{
      Group{
        Name
      }
    }
  }
}
```
And I wanted it to be filtered like Name of Student contains "John" and Group Name is "Group1"
So I've tried to do like inside of schema to make for first an include query
```ts
students2: (_parent, _args, context) => {
      return context.prisma.people_List.findMany({
        where:{
          Name: {
            contains: "John"
          }
        },
        include:{
          Status: {
            Group: {
              Name: "Group-1"
            }
          }
        }
      })
    },
  },
```
But then I've received an error like that?
```
"message": "\nInvalid `prisma.people_List.findMany()` invocation:\n\n{\n  where: {\n    Name: {\n      contains: 'John'\n    }\n  },\n  include: {\n    Status: {\n    ~~~~~~\n      Group: {\n        Name: 'Group-1'\n      }\n    },\n?   spr_Nationality?: true\n  }\n}\n\n\nUnknown field `Status` for include statement on model people_List. Available options are listed in green.\n"
```
Is it possible even to make deep nest include?

**Comment by ryands17** [comment]
Do you need to find all people with name `John` and then filter the group or do you want to perform an `and` condition?

**Reply by BJladika** [reply]
I'd like to `and` condition, aka find all people with name `John` in `Group1`

**Reply by ryands17** [reply]
Could you share your `schema.prisma` as well with the relation?

**Reply by BJladika** [reply]
Like that?
```graphql
type Student {
  id_People: ID!
  Status: [Status]
}
type Status{
  id_PeoplePln: ID!
  id_People: ID
  Group: Group
  id_Group: ID
}
type Group{
  id_Group: ID!
  Name: String
  Students: [Status!]!
}
```
```prisma
model people_Pln {
  id_PeoplePln        Int            @id
  id_People           Int
  id_Group            Int
  pln_group_Stud      pln_group_Stud @relation(fields: [id_Group], references: [id_Group])
}
model people_List {
  id_People           Int              @id
  Surname             String           @db.VarChar(30)
  Name                String           @db.VarChar(30)
  Patronymic          String?          @db.VarChar(30)
  id_Nationality      Int?
  spr_Nationality     spr_Nationality? @relation(fields: [id_Nationality], references: [id_Nationality])
}
model pln_group_Stud {
  id_Group        Int          @id
  Name            String       @db.VarChar(30)
  people_Pln      people_Pln[]
}
```
In short I wanted make a nested query similar to SQL query like
```
SELECT * from 
people_List as PL, 
people_Pln as PP, 
pln_group_Stud as G 
where G.Name Like '%Group1'
And
G.id_Group = PP.id_Group
and
PP.id_People = PL.id_People
and
PL.Name like 'John'
```

Ah, I also wanted to ask, I'm trying to make backward query like
```graphql
query{
  groups{
    Students{
      id_Group
      id_People
      Info{
        id_People
        Name
    }
  }
}

```

and tried to make resolver do findUnique (similar to finding posts of user, like user{posts{name}})
```prisma
  Group:{
    Students: (parent, _args, ctx) => ctx.prisma.people_Pln.findUnique({ where: { id_Group: parent.id_Group}}).people_Pln(),
  },
```
but still receiving that `people_Pln()` is not a function, what exactly I need to write

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
